wireapp/wire-ios-mono
# feat: improve backend confirmation alert WPB-4242

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-4242" title="WPB-4242" target="_blank"><img alt="Story" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10815?size=medium" />WPB-4242</a>  [iOS] Misleading backend enrollment dialogue
  </td></table>
  <br />
 









































### Issue

When switching to a new backend we first ask for user confirmation if they want to connect to a new backend. This alert is very basic and contains only the backend name and url. Switching to a backend involves communication to several urls, which are not shown to the user in this alert.

In order to improve the transparency to the user, we now show all the URLs of the new backend in new confirmation modal. This confirmation is shown both when the tries to connect to a new backend via a deep link as well as then logging in via SSO.

The new screen is written in SwiftUI, and as such I've also implemented some helpers, namely defining the button styles and text styles according to the design system.

### Note
I tried to improve the code for switching backends but it snowballed into a lot of changes and got a bit messy, so instead I opted to just make the necessary changes without rewriting everything. 

### Testing

1. Tap on a deep link to switch backends
2. Verify that you see the confirmation.
3. Verify if you tap cancel, the backend is not switched.
4. Verify if you tap proceed, the backend is switched.

---

### Checklist
